doc/modules/calibration.rst

```diff
@@ -74,10 +74,14 @@ by showing the number of samples in each predicted probability bin.
 
 .. currentmodule:: sklearn.linear_model
 
-:class:`LogisticRegression` returns well calibrated predictions by default as it has a
+:class:`LogisticRegression` is more likely to return well calibrated predictions by itself as it has a
 canonical link function for its loss, i.e. the logit-link for the :ref:`log_loss`.
-This leads to the so-called **balance property**, see [8]_ and
-:ref:`Logistic_regression`.
+In the unpenalized case, this leads to the so-called **balance property**, see [8]_ and :ref:`Logistic_regression`.
```

Comment:
I read the reference that I found interesting. However, the balance property only involves the sample mean prediction (a single scalar) not the 10 bucketed mean predictions of the reliability curve. So that property alone, does not explain why LR tends to be better calibrated than other models in this study.

Reply:
Note that the balance property holds for all features $j$, not only for the intercept, i.e. $0 = \sum_i x_{i,j} (y_i - m(x_{i,j}))$. In a way, that relates to conditional calibration $m(X) = E[Y|X]$. Conditional calibration implies auto-calibration $m(X) = E[Y|m(X)]$.

Reply:
The balanced property does imply (even in-sample) auto-calibration on the dataset as demonstrated empirically on a dataset generated in a way that makes Logistic Regression a misspecified model, see:

https://github.com/scikit-learn/scikit-learn/pull/28171#discussion_r1464660588

Reply:
Let's close this to keep the PR focused on the original problem.